File: CP/CF/CF_round_733_div12_alt/5.py
import os, sys, bisect
from collections import defaultdict, Counter, deque;
from functools import lru_cache   #use @lru_cache(None)
if os.path.exists('in.txt'): sys.stdin=open('in.txt','r')
if os.path.exists('out.txt'): sys.stdout=open('out.txt', 'w')
#
input = lambda: sys.stdin.readline().strip()
imap = lambda: map(int,input().split()); ilist = lambda: list(imap())
#------------------------------------------------------------------
#sys.setrecursionlimit(10**6)
mod = int(1e9+7)


class Person:
    def __init__(self, totalAccountBalance=0, age=0, salary=0, numOfDependents=0):
        self.totalAccountBalance = totalAccountBalance
        self.age = age
        self.salary = salary
        self.numOfDependents = numOfDependents

    def keyToCompare(self):
        return [self.totalAccountBalance, self.age, self.salary, self.numOfDependents]

    def __repr__(self):
        return str(self.keyToCompare())

listOfPersons = [Person(0,1,1,2), Person(5,2,5,2), Person(2,42,4,2), Person(0,3,4,2)]

listOfPersons.sort(key = lambda x: x.keyToCompare())
print(listOfPersons)

File: CP/CF/CF_round_733_div12_alt/5.java
import java.util.*;

// Person.java
public class Person implements Comparable<Person>{
    public int totalAccountBalance;
    public int age;
    public int salary;
    public int numOfDependents;

    public Person(int totalAccountBalance,int age, int salary, int numOfDependents){
        this.totalAccountBalance = totalAccountBalance;
        this.age = age;
        this.salary = salary;
        this.numOfDependents = numOfDependents;
    }

    @Override
    public int compareTo(Person other){
        if (this.totalAccountBalance-other.totalAccountBalance!=0)
            return this.totalAccountBalance-other.totalAccountBalance;
        if (this.age-other.age!=0)
            return this.age-other.age;
        if (this.salary-other.salary!=0)
            return this.salary-other.salary;
        return this.numOfDependents - other.numOfDependents;
    }

    @Override
    public String toString(){
        return String.format("total Account Balance: %s, age: %s, salary: %s, number Of Dependents: %s", totalAccountBalance, age, salary, numOfDependents);
    }
}


// Main.java
import java.util.*;
public class Main{
    public static void main(String[] args){
        List<Person> listOfPersons = new ArrayList<Person>();
        listOfPersons.add(new Person(0,1,1,2));
        listOfPersons.add(new Person(5,2,5,2));
        listOfPersons.add(new Person(2,42,4,2));
        listOfPersons.add(new Person(0,3,4,2));

        Collections.sort(listOfPersons);

        for(Person person: listOfPersons){
            System.out.println(person);
        }

    }
}
